### PR TITLE
feat(CR): Add a new Clearing Request state PENDING INPUT

### LINKED ISSUE #2374: Update Clearing Request State from "Awaiting Response" to "Pending Input"
Currently the clearing request state is set to "**Awaiting Response**" when the clearing team is waiting for input/action from the requesting user. This needs to changed to **"Pending Input**" as this is more intuitive for the user and makes it clear that something is pending from the requester's end.
pending_input_pr

### PR DESCRIPTION
**Pull request by akshitjoshii**
Issue:Closes #2374 
Changes the Clearing status from "Awaiting Response" to "Pending Input" as it requires input from the CR requester and is more intuitive for the user.
 
pending_input_pr2

### CODE REVIEW

File: frontend/sw360-portlet/src/main/resources/content/Language.properties [hunk 1]
```diff
@@ -116,7 +116,7 @@ authorities=Authorities
 authorization.header.authorization.token.api.token=Authorization Header (Authorization: Token <API-Token>)
 auto.refresh.in=Auto-refresh in
 auto-refresh.in.5=Auto-refresh in 5
-awaiting.response=Awaiting Response
+pending.input=Pending Input
```
Comment (afsahsyeda) [inline frontend/sw360-portlet/src/main/resources/content/Language.properties:119]: This should come after p in alphabetical order

File: frontend/sw360-portlet/src/main/resources/content/Language_ja.properties [hunk 2]
```diff
@@ -116,7 +116,7 @@ authorities=権利者
 authorization.header.authorization.token.api.token=認証ヘッダ, (認証 API-トークン)
 auto.refresh.in=自動更新
 auto-refresh.in.5=5分毎に自動更新
-awaiting.response=Awaiting Response
+pending.input=Pending Input
```
Comment (afsahsyeda) [inline frontend/sw360-portlet/src/main/resources/content/Language_ja.properties:119]: maintain alphabetical order

File: frontend/sw360-portlet/src/main/resources/content/Language_vi.properties [hunk 3]
```diff
@@ -116,7 +116,7 @@ authorities=Quyền
 authorization.header.authorization.token.api.token=Phần đầu mã truy cập. (Authorization: Mã truy cập <API-Token>)
 auto.refresh.in=Tự động làm mới trong
 auto-refresh.in.5=Tự động làm mới trong 5
-awaiting.response=Awaiting Response
+pending.input=Pending Input
```
Comment (afsahsyeda) [inline frontend/sw360-portlet/src/main/resources/content/Language_vi.properties:119]: maintain alphabetical order

File: frontend/sw360-portlet/src/main/resources/content/Language_zh.properties [hunk 4]
```diff
@@ -116,7 +116,7 @@ authorities=权利人
 authorization.header.authorization.token.api.token=授权标头（Authorization: Token <API-Token>）
 auto.refresh.in=自动刷新
 auto-refresh.in.5=每5分钟自动刷新
-awaiting.response=Awaiting Response
+pending.input=Pending Input
```
Comment (afsahsyeda) [inline frontend/sw360-portlet/src/main/resources/content/Language_zh.properties:119]: maintain alphabetical order

**Review by afsahsyeda — CHANGES_REQUESTED** [review]

**Comment by sameed20** [comment]
Testing this PR.

**Comment by akshitjoshii** [comment]
Addressed review comments.

**Review by heliocastro — COMMENTED** [review]
Is this not a HUGE breaking change ?
One thing is change the messages, other is change the id of a database enum.
Do we have automated migration scripts ?

@KoukiHama How is this related to Toshiba ?

**Comment by afsahsyeda** [comment]
> Is this not a HUGE breaking change ? One thing is change the messages, other is change the id of a database enum. Do we have automated migration scripts ?
> 
> @KoukiHama How is this related to Toshiba ?

Hi @heliocastro , the value of the enum remains the same on the backend. Only the term has been updated to "PENDING_INPUT" to align with its corresponding title that states "Clearing Team is waiting for input/action from the Requesting User".

**Comment by sameed20** [comment]
Hi @akshitjoshii After these changes the CR's which are created before introducing this feature are not loading.

**Comment by akapti** [comment]
> Hi @akshitjoshii After these changes the CR's which are created before introducing this feature are not loading.

For old CR's to load, One should write database migration script, like @heliocastro suggested.
What's the requirement @akshitjoshii ?
Is it to replace the existing entry `Awaiting response` with new entry `Pending Input`? or just to add new entry `Pending Input`?

If it's replacing then you need to add database script for migrating old CR's in `Awaiting response` state.

**Comment by akshitjoshii** [comment]
Hi @akapti as per the new requirement a new field called "PENDING INPUT" is to be added keeping the existing entry "AWAITING RESPONSE" as it is. Then a migration script won't be needed.

**Comment by akshitjoshii** [comment]
As per suggestion, a new state PENDING INPUT has been added keeping the existing state AWAITING RESPONSE as it is.

File: frontend/sw360-portlet/src/main/resources/content/Language.properties [hunk 5]
```diff
@@ -210,14 +210,16 @@ ClearingRequestPriority-CRITICAL=Clearing time less than 1 week.
...
+ClearingRequestState=New: CR is pending to be Accepted by the Clearing Team.&#10A sanity check for the artifacts required for Clearing will be done prior to Accepting a CR. &#10Pending Input: Inputs to perform clearing is incomplete. Clearing team cannot&#10process the CR to next stage until project provides the missing inputs.&#10Sanity Check: The Clearing team is performing the sanity check on inputs provided by project team.&#10Source code availability for all linked releases are checked during this stage. &#10Accepted: CR has passed SanityCheck and is accepted into the clearing queue. &#10Rejected: CR has failed SanityCheck. Contact Clearing Team. (Rejecting resets the CR priority) &#10In Queue: CR is next in queue for clearing. &#10In Progress: CR is being worked on by the clearing team. &#10Closed: Clearings requested in the CR is complete. (Closing resets the CR priority) &#10Awaiting Response: Clearing Team is waiting for input/action from the Requesting User.
```
Comment (afsahsyeda) [inline frontend/sw360-portlet/src/main/resources/content/Language.properties:213]: Pending Input: 
The input needed for clearing is incomplete. The clearing team cannot proceed to the next stage of processing the CR until the project provides the missing input.
Sanity Check:
The Clearing team is conducting a sanity check on the inputs provided by the project team. At this stage, they verify the availability of the source code for all linked releases

Please change the titles accordingly.

File: frontend/sw360-portlet/src/main/resources/content/Language_ja.properties [hunk 6]
```diff
@@ -209,14 +209,16 @@ ClearingRequestPriority-CRITICAL=クリアリング時間が1週以内.
...
 ClearingRequestState-NEW=保留されていたクリアリングリクエストがクリアリングチームによって承認.&#10A クリアリングに必要な成果物のSanityCheckはクリアリングリクエスト承認前に行う.
 ClearingRequestState-ACCEPTED=クリアリングリクエストはSanityCheckが完了しクリアリング受領.
 ClearingRequestState-REJECTED=クリアリングリクエストはSanityCheckに失敗. クリアリングチームに連絡してください. (Rejecting resets the CR priority)
 ClearingRequestState-IN_QUEUE=クリアリングリクエストが次のリクエストに入りました．
 ClearingRequestState-IN_PROGRESS=クリアリングリクエストはクリアリングチームに確認されています．
 ClearingRequestState-CLOSED=クリアリングリクエスト完了． (Closing resets the CR priority)
 ClearingRequestState-AWAITING_RESPONSE=クリアリングチームはユーザーからの対応待ちです．
+ClearingRequestState-PENDING_INPUT=Clearing Team is waiting for input/action from the Requesting User.
+ClearingRequestState-SANITY_CHECK = A sanity check for the artifacts required for Clearing will be done prior to Accepting a CR
```
Comment (afsahsyeda) [inline frontend/sw360-portlet/src/main/resources/content/Language_ja.properties:221]: ClearingRequestState-SANITY_CHECK = Prior to accepting a CR, a sanity check for the artifacts required for Clearing will be performed.

**Review by afsahsyeda — CHANGES_REQUESTED** [review]

**Comment by sameed20** [comment]
Test Successful.
pr_testing
pr_testing_3

**Comment by sameed20** [comment]
Test Successful.
new_testing_1
new_testing_2
new_testing_3

**Comment by afsahsyeda** [comment]
Code looks good to me.

**Review by afsahsyeda — APPROVED** [review]

**Pull request merged**